site/docs/v1/tech/oauth/integrations/salesforce.adoc

```diff
@@ -0,0 +1,213 @@
+---
+layout: doc
+title: "OAuth Integrations: Salesforce"
+description: A practical approach of using FusionAuth SSO capabilities as a login provider in a Salesforce application
+navcategory: login-methods
+---
+
+== Overview
+
+link:https://www.salesforce.com/[Salesforce] is a robust customer relationship management software that allows users to sign in into their accounts using an external Identity Provider. This document covers the configuration necessary to get Salesforce working with FusionAuth as the identity provider using OpenID Connect.
+
+include::docs/v1/tech/samlv2/_saml_idp_note.adoc[]
+
+=== Prerequisites
+
+This document assumes you have a running instance of FusionAuth and a working Salesforce application. You will also need admin accounts for both to configure them correctly and a FusionAuth user that you will use to sign into Salesforce. You can use an existing user or create a new user for this purpose.
+
+== Configure FusionAuth
+
+Navigate to [breadcrumb]#Applications# and click icon:plus[role=ui-button green,type=fas] to create an application named `Salesforce`.
+
+image::core-concepts/create-application.png[Create an Application,width=1200]
+
+Click icon:save[role=ui-button blue,type=fas] to save your application. After being redirected to the Applications page, click on the icon:search[role=ui-button green,type=fas] icon to view the application details. Write down the [field]#OAuth IdP login URL#, [field]#Logout URL#, [field]#Token endpoint# and [field]#Userinfo endpoint# fields, located in the [uielement]#OAuth2 & OpenID Connect Integration details# section, as you'll need that information later.
+
+== Configure Salesforce
+
+Before configuring FusionAuth as an identity provider in Salesforce, you need to create some code that will map the information returned from FusionAuth into a user entity that will then be persisted in Salesforce using  link:https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_intro_what_is_apex.htm[Apex], their object-oriented programming language.
+
+Navigate to [breadcrumb]#PLATFORM TOOLS -> Custom Code -> Apex Classes# and click the [uielement]#New# button to create a new class.
+
+image::oauth/integrations/salesforce/apex-class-new.png[Create an Apex class.,width=1276,role=bottom-cropped]
+
+In the editor, paste the following code:
+
+```java
```

Comment:
Instead of backticks, use a structure like this:

```
[source,java]
----
<sample code here>
----
```

Is Apex close enough to Java that it highlights nicely? I'm not sure what all the options are, but I assume they don't support Apex. If Java highlighting is very strange, we may be better off not specifying a source language.

Reply:
The only thing that is kind of strange is how single quotes are highlighted (and Apex doesn't support double quotes):

![image](https://github.com/FusionAuth/fusionauth-site/assets/1877191/95315777-a03c-49d2-b762-a9e5d4cba1e7)

Besides that, I think it's pretty good.